coriolis/tests/osmorphing/test_ubuntu.py

```diff
@@ -0,0 +1,176 @@
+# Copyright 2024 Cloudbase Solutions Srl
+# All Rights Reserved.
+
+import logging
+from unittest import mock
+
+from coriolis.osmorphing import base
+from coriolis.osmorphing import ubuntu
+from coriolis.tests import test_base
+
+
+class BaseUbuntuMorphingToolsTestCase(test_base.CoriolisBaseTestCase):
+    """Test suite for the BaseUbuntuMorphingTools class."""
+
+    def setUp(self):
+        super(BaseUbuntuMorphingToolsTestCase, self).setUp()
+        self.detected_os_info = {
+            'os_type': 'linux',
+            'distribution_name': 'Ubuntu',
+            'release_version': '20.04',
+            'friendly_release_name': mock.sentinel.friendly_release_name,
+        }
+        self.event_manager = mock.MagicMock()
+        self.os_root_dir = '/root'
+        self.morphing_tools = ubuntu.BaseUbuntuMorphingTools(
+            mock.sentinel.conn, self.os_root_dir,
+            mock.sentinel.os_root_dev, mock.sentinel.hypervisor,
+            self.event_manager, self.detected_os_info,
+            mock.sentinel.osmorphing_parameters,
+            mock.sentinel.operation_timeout)
+
+    @mock.patch.object(
+        base.BaseLinuxOSMorphingTools, '_version_supported_util'
+    )
+    def test_check_os_supported_not_supported(
+            self, mock_version_supported_util):
+        self.detected_os_info['distribution_name'] = 'unsupported'
+
+        result = ubuntu.BaseUbuntuMorphingTools.check_os_supported(
+            self.detected_os_info)
+
+        mock_version_supported_util.assert_not_called()
+
+        self.assertFalse(result)
+
+    @mock.patch.object(
+        base.BaseLinuxOSMorphingTools, '_version_supported_util'
+    )
+    def test_check_os_supported_lts_release(self, mock_version_supported_util):
+        result = ubuntu.BaseUbuntuMorphingTools.check_os_supported(
+            self.detected_os_info)
+
+        mock_version_supported_util.assert_called_once_with(
+            self.detected_os_info['release_version'], minimum=12.04,
+            maximum=12.04)
+
+        self.assertTrue(result)
+
+    @mock.patch.object(
+        base.BaseLinuxOSMorphingTools, '_version_supported_util'
+    )
+    def test_check_os_supported_non_lts_release(
+            self, mock_version_supported_util):
+        self.detected_os_info['release_version'] = '21.10'
+        mock_version_supported_util.return_value = False
+
+        result = ubuntu.BaseUbuntuMorphingTools.check_os_supported(
+            self.detected_os_info)
+
+        mock_version_supported_util.assert_called_with(
+            self.detected_os_info['release_version'], minimum=22.04)
+
+        self.assertEqual(result, mock_version_supported_util.return_value)
+
+    @mock.patch('yaml.dump')
```

Comment:
The test can be written without mocking this, but you have to adjust the inputs / assertions.